From b93b092d6e7f27c9410d60f07c3576b6a74ff75f Mon Sep 17 00:00:00 2001
From: Lance Albertson <lance@osuosl.org>
Date: Sat, 5 Oct 2019 13:00:14 -0700
Subject: [PATCH 2/4] Cinc omnibus patches

This pulls in changes that were originally in the client repository. The idea is
that we can easily rebase the changes here from upstream and merge any future
conflicts more easily.

Some other notable changes:

- Update omnibus Gemfile to use a patched version of omnibus-software which includes
  changes in the chef software and adds our rebranded chef-zero
- Add support for Angry Cinc Client (useful for our build infrastructure)
- Update Copyright to add "Cinc Project" to attribute our edits
---
 chef-bin/bin/{chef-apply => cinc-apply}       |  2 +-
 chef-bin/bin/{chef-client => cinc-client}     |  2 +-
 ...urce-inspector => cinc-resource-inspector} |  0
 ...f-service-manager => cinc-service-manager} |  2 +-
 chef-bin/bin/{chef-shell => cinc-shell}       |  0
 chef-bin/bin/{chef-solo => cinc-solo}         |  2 +-
 ...f-windows-service => cinc-windows-service} |  2 +-
 omnibus/Gemfile                               |  2 +-
 .../projects/{angrychef.rb => angrycinc.rb}   | 15 +++++-----
 omnibus/config/projects/{chef.rb => cinc.rb}  | 29 ++++++++++---------
 .../package-scripts/{chef => cinc}/postinst   | 29 ++++++++++++-------
 omnibus/package-scripts/{chef => cinc}/postrm | 11 ++++++-
 .../package-scripts/{chef => cinc}/preinst    |  6 ++--
 13 files changed, 61 insertions(+), 41 deletions(-)
 rename chef-bin/bin/{chef-apply => cinc-apply} (93%)
 rename chef-bin/bin/{chef-client => cinc-client} (93%)
 rename chef-bin/bin/{chef-resource-inspector => cinc-resource-inspector} (100%)
 rename chef-bin/bin/{chef-service-manager => cinc-service-manager} (95%)
 rename chef-bin/bin/{chef-shell => cinc-shell} (100%)
 rename chef-bin/bin/{chef-solo => cinc-solo} (93%)
 rename chef-bin/bin/{chef-windows-service => cinc-windows-service} (93%)
 rename omnibus/config/projects/{angrychef.rb => angrycinc.rb} (79%)
 rename omnibus/config/projects/{chef.rb => cinc.rb} (79%)
 rename omnibus/package-scripts/{chef => cinc}/postinst (70%)
 rename omnibus/package-scripts/{chef => cinc}/postrm (74%)
 rename omnibus/package-scripts/{chef => cinc}/preinst (86%)

diff --git a/chef-bin/bin/chef-apply b/chef-bin/bin/cinc-apply
similarity index 93%
rename from chef-bin/bin/chef-apply
rename to chef-bin/bin/cinc-apply
index 05b975a11..e07240725 100755
--- a/chef-bin/bin/chef-apply
+++ b/chef-bin/bin/cinc-apply
@@ -21,4 +21,4 @@
 $:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require "chef/application/apply"
 
-Chef::Application::Apply.new.run(enforce_license: true)
+Chef::Application::Apply.new.run(enforce_license: false)
diff --git a/chef-bin/bin/chef-client b/chef-bin/bin/cinc-client
similarity index 93%
rename from chef-bin/bin/chef-client
rename to chef-bin/bin/cinc-client
index 45a6af546..95402c948 100755
--- a/chef-bin/bin/chef-client
+++ b/chef-bin/bin/cinc-client
@@ -22,4 +22,4 @@ $:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require "chef"
 require "chef/application/client"
 
-Chef::Application::Client.new.run(enforce_license: true)
+Chef::Application::Client.new.run(enforce_license: false)
diff --git a/chef-bin/bin/chef-resource-inspector b/chef-bin/bin/cinc-resource-inspector
similarity index 100%
rename from chef-bin/bin/chef-resource-inspector
rename to chef-bin/bin/cinc-resource-inspector
diff --git a/chef-bin/bin/chef-service-manager b/chef-bin/bin/cinc-service-manager
similarity index 95%
rename from chef-bin/bin/chef-service-manager
rename to chef-bin/bin/cinc-service-manager
index 64cc043a5..d8c9877a0 100755
--- a/chef-bin/bin/chef-service-manager
+++ b/chef-bin/bin/cinc-service-manager
@@ -34,5 +34,5 @@ if Chef::Platform.windows?
   }
   Chef::Application::WindowsServiceManager.new(chef_client_service).run
 else
-  puts "chef-service-manager is only available on Windows platforms."
+  puts "cinc-service-manager is only available on Windows platforms."
 end
diff --git a/chef-bin/bin/chef-shell b/chef-bin/bin/cinc-shell
similarity index 100%
rename from chef-bin/bin/chef-shell
rename to chef-bin/bin/cinc-shell
diff --git a/chef-bin/bin/chef-solo b/chef-bin/bin/cinc-solo
similarity index 93%
rename from chef-bin/bin/chef-solo
rename to chef-bin/bin/cinc-solo
index 7a2168230..3530411ac 100755
--- a/chef-bin/bin/chef-solo
+++ b/chef-bin/bin/cinc-solo
@@ -21,4 +21,4 @@
 $:.unshift(File.join(File.dirname(__FILE__), "..", "lib"))
 require "chef/application/solo"
 
-Chef::Application::Solo.new.run(enforce_license: true)
+Chef::Application::Solo.new.run(enforce_license: false)
diff --git a/chef-bin/bin/chef-windows-service b/chef-bin/bin/cinc-windows-service
similarity index 93%
rename from chef-bin/bin/chef-windows-service
rename to chef-bin/bin/cinc-windows-service
index 646e3b4a9..f48db8b19 100755
--- a/chef-bin/bin/chef-windows-service
+++ b/chef-bin/bin/cinc-windows-service
@@ -30,5 +30,5 @@ require "chef/application/windows_service"
 if Chef::Platform.windows?
   Chef::Application::WindowsService.mainloop
 else
-  puts "chef-windows-service is only available on Windows platforms."
+  puts "chef-cinc-service is only available on Windows platforms."
 end
diff --git a/omnibus/Gemfile b/omnibus/Gemfile
index 6fca425e6..5f364a8f9 100644
--- a/omnibus/Gemfile
+++ b/omnibus/Gemfile
@@ -1,7 +1,7 @@
 source "https://rubygems.org"
 
 gem "omnibus", git: "https://github.com/chef/omnibus", branch: "master"
-gem "omnibus-software", git: "https://github.com/chef/omnibus-software", branch: "master"
+gem "omnibus-software", path: "../../omnibus-software"
 gem "artifactory"
 
 gem "pedump"
diff --git a/omnibus/config/projects/angrychef.rb b/omnibus/config/projects/angrycinc.rb
similarity index 79%
rename from omnibus/config/projects/angrychef.rb
rename to omnibus/config/projects/angrycinc.rb
index ff83f4132..7d15712a5 100644
--- a/omnibus/config/projects/angrychef.rb
+++ b/omnibus/config/projects/angrycinc.rb
@@ -1,5 +1,6 @@
 #
 # Copyright 2012-2016, Chef Software, Inc.
+# Copyright 2019, Cinc Project
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -20,23 +21,23 @@
 # `config/project/chef.rb`.
 #
 current_file = __FILE__
-chef_project_contents = IO.read(File.expand_path("../chef.rb", __FILE__))
+chef_project_contents = IO.read(File.expand_path("../cinc.rb", __FILE__))
 instance_eval chef_project_contents
 
-name "angrychef"
-friendly_name "Angry Chef Client"
+name "angrycinc"
+friendly_name "Angry Cinc Client"
 
 if windows?
   # NOTE: Ruby DevKit fundamentally CANNOT be installed into "Program Files"
   #       Native gems will use gcc which will barf on files with spaces,
   #       which is only fixable if everyone in the world fixes their Makefiles
-  install_dir "#{default_root}/opscode/#{name}"
-  package_name "angrychef"
+  install_dir "#{default_root}/cinc-project/#{name}"
+  package_name "angrycinc"
 else
   install_dir "#{default_root}/#{name}"
 end
 
-resources_path "#{resources_path}/../chef"
+resources_path "#{resources_path}/../cinc"
 
 msi_upgrade_code = "D7FDDC1A-7668-404E-AD2F-61F875632A9C"
-project_location_dir = "angrychef"
+project_location_dir = "angrycinc"
diff --git a/omnibus/config/projects/chef.rb b/omnibus/config/projects/cinc.rb
similarity index 79%
rename from omnibus/config/projects/chef.rb
rename to omnibus/config/projects/cinc.rb
index b3768a58a..f2a534ff4 100644
--- a/omnibus/config/projects/chef.rb
+++ b/omnibus/config/projects/cinc.rb
@@ -1,5 +1,6 @@
 #
 # Copyright 2012-2018, Chef Software Inc.
+# Copyright 2019, Cinc Project
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.
@@ -14,12 +15,12 @@
 # limitations under the License.
 #
 
-name "chef"
-friendly_name "Chef Infra Client"
-maintainer "Chef Software, Inc. <maintainers@chef.io>"
-homepage "https://www.chef.io"
-license "Chef EULA"
-license_file "CHEF-EULA.md"
+name "cinc"
+friendly_name "Cinc Client"
+maintainer "Cinc Project <maintainers@cinc.sh>"
+homepage "https://www.cinc.sh"
+license "Apache-2.0"
+license_file "../LICENSE"
 
 build_iteration 1
 # Do not use __FILE__ after this point, use current_file. If you use __FILE__
@@ -33,8 +34,8 @@ if windows?
   # NOTE: Ruby DevKit fundamentally CANNOT be installed into "Program Files"
   #       Native gems will use gcc which will barf on files with spaces,
   #       which is only fixable if everyone in the world fixes their Makefiles
-  install_dir  "#{default_root}/opscode/#{name}"
-  package_name "chef-client"
+  install_dir  "#{default_root}/cinc-project/#{name}"
+  package_name "cinc"
 else
   install_dir "#{default_root}/#{name}"
 end
@@ -71,7 +72,7 @@ end
 dependency "ruby-cleanup"
 
 package :rpm do
-  signing_passphrase ENV["OMNIBUS_RPM_SIGNING_PASSPHRASE"]
+  # signing_passphrase ENV["OMNIBUS_RPM_SIGNING_PASSPHRASE"]
   compression_level 1
   compression_type :xz
 end
@@ -83,8 +84,8 @@ end
 
 proj_to_work_around_cleanroom = self # wat voodoo hackery is this?
 package :pkg do
-  identifier "com.getchef.pkg.#{proj_to_work_around_cleanroom.name}"
-  signing_identity "Developer ID Installer: Chef Software, Inc. (EU3VF8YLX2)"
+  identifier "com.cinc-project.pkg.#{proj_to_work_around_cleanroom.name}"
+  # signing_identity "Developer ID Installer: Chef Software, Inc. (EU3VF8YLX2)"
 end
 compress :dmg
 
@@ -95,11 +96,11 @@ package :msi do
   upgrade_code msi_upgrade_code
   wix_candle_extension "WixUtilExtension"
   wix_light_extension "WixUtilExtension"
-  signing_identity "AF21BA8C9E50AE20DA9907B6E2D4B0CC3306CA03", machine_store: true
-  parameters ChefLogDllPath: windows_safe_path(gem_path("chef-[0-9]*-mingw32/ext/win32-eventlog/chef-log.dll")),
+  # signing_identity "AF21BA8C9E50AE20DA9907B6E2D4B0CC3306CA03", machine_store: true
+  parameters CincLogDllPath: windows_safe_path(gem_path("chef-[0-9]*-mingw32/ext/win32-eventlog/chef-log.dll")),
              ProjectLocationDir: project_location_dir
 end
 
 package :appx do
-  signing_identity "AF21BA8C9E50AE20DA9907B6E2D4B0CC3306CA03", machine_store: true
+#  signing_identity "AF21BA8C9E50AE20DA9907B6E2D4B0CC3306CA03", machine_store: true
 end
diff --git a/omnibus/package-scripts/chef/postinst b/omnibus/package-scripts/cinc/postinst
similarity index 70%
rename from omnibus/package-scripts/chef/postinst
rename to omnibus/package-scripts/cinc/postinst
index c1c527290..6d69637d7 100755
--- a/omnibus/package-scripts/chef/postinst
+++ b/omnibus/package-scripts/cinc/postinst
@@ -11,8 +11,8 @@
 #
 
 PROGNAME=`basename $0`
-INSTALLER_DIR=/opt/chef
-CONFIG_DIR=/etc/chef
+INSTALLER_DIR=/opt/cinc
+CONFIG_DIR=/etc/cinc
 USAGE="usage: $0 [-v validation_key] ([-o organization] || [-u url])"
 
 error_exit()
@@ -87,25 +87,34 @@ rm -f $PREFIX/bin/chef-shell
 rm -f $PREFIX/bin/knife
 rm -f $PREFIX/bin/ohai
 
-ln -sf $INSTALLER_DIR/bin/chef-solo $PREFIX/bin || error_exit "Cannot link chef-solo to $PREFIX/bin"
-if [ -f "$INSTALLER_DIR/bin/chef-apply" ]; then
-  ln -sf $INSTALLER_DIR/bin/chef-apply $PREFIX/bin || error_exit "Cannot link chef-apply to $PREFIX/bin"
+ln -sf $INSTALLER_DIR/bin/cinc-solo $PREFIX/bin || error_exit "Cannot link cinc-solo to $PREFIX/bin"
+if [ -f "$INSTALLER_DIR/bin/cinc-apply" ]; then
+  ln -sf $INSTALLER_DIR/bin/cinc-apply $PREFIX/bin || error_exit "Cannot link cinc-apply to $PREFIX/bin"
 fi
-if [ -f "$INSTALLER_DIR/bin/chef-shell" ]; then
-  ln -sf $INSTALLER_DIR/bin/chef-shell $PREFIX/bin || error_exit "Cannot link chef-shell to $PREFIX/bin"
+if [ -f "$INSTALLER_DIR/bin/cinc-shell" ]; then
+  ln -sf $INSTALLER_DIR/bin/cinc-shell $PREFIX/bin || error_exit "Cannot link cinc-shell to $PREFIX/bin"
 fi
 ln -sf $INSTALLER_DIR/bin/knife $PREFIX/bin || error_exit "Cannot link knife to $PREFIX/bin"
 ln -sf $INSTALLER_DIR/bin/ohai $PREFIX/bin || error_exit "Cannot link ohai to $PREFIX/bin"
 
-# We test for the presence of /usr/bin/chef-client to know if this script succeeds, so this
+# We test for the presence of /usr/bin/cinc-client to know if this script succeeds, so this
 # must appear as the last real action in the script
-ln -sf $INSTALLER_DIR/bin/chef-client $PREFIX/bin || error_exit "Cannot link chef-client to $PREFIX/bin"
+ln -sf $INSTALLER_DIR/bin/cinc-client $PREFIX/bin || error_exit "Cannot link chef-client to $PREFIX/bin"
+
+wrapper_links="chef-apply chef-client chef-shell chef-solo"
+link_target="cinc-wrapper"
+for link in $wrapper_links; do
+  if [ ! -e $PREFIX/bin/$link ]; then
+    echo "Symlinking $link command to cinc-wrapper for compatibility..."
+    ln -sf $INSTALLER_DIR/bin/$link_target $PREFIX/bin/$link || error_exit "Cannot link $link_target to $PREFIX/bin/$link"
+  fi
+done
 
 # Ensure all files/directories in $INSTALLER_DIR are owned by root. This
 # has been fixed on new installs but upgrades from old installs need to
 # be manually fixed.
 chown -Rh 0:0 $INSTALLER_DIR
 
-echo "Thank you for installing Chef Infra Client! For help getting started visit https://learn.chef.io"
+echo "Thank you for installing Cinc Client, the community build based on Chef Infra Client!"
 
 exit 0
diff --git a/omnibus/package-scripts/chef/postrm b/omnibus/package-scripts/cinc/postrm
similarity index 74%
rename from omnibus/package-scripts/chef/postrm
rename to omnibus/package-scripts/cinc/postrm
index a153da710..7a0905f09 100755
--- a/omnibus/package-scripts/chef/postrm
+++ b/omnibus/package-scripts/cinc/postrm
@@ -33,10 +33,19 @@ else
 fi
 
 cleanup_symlinks() {
-  binaries="chef-client chef-solo chef-apply chef-shell knife ohai"
+  binaries="cinc-client cinc-solo cinc-apply cinc-shell knife ohai"
   for binary in $binaries; do
     rm -f $PREFIX/bin/$binary
   done
+
+  wrapper_links="chef-apply chef-client chef-shell chef-solo"
+  link_target="cinc-wrapper"
+  for link in $wrapper_links;  do
+    if [ -L $PREFIX/bin/$link -a "$(readlink $PREFIX/bin/$link)" = "$INSTALLER_DIR/bin/$link_target" ]; then
+      echo "Removing $link compatibility symlink..."
+      rm -f $PREFIX/bin/$link
+    fi
+  done
 }
 
 # Clean up binary symlinks if they exist
diff --git a/omnibus/package-scripts/chef/preinst b/omnibus/package-scripts/cinc/preinst
similarity index 86%
rename from omnibus/package-scripts/chef/preinst
rename to omnibus/package-scripts/cinc/preinst
index e38c3f7a3..86548c869 100755
--- a/omnibus/package-scripts/chef/preinst
+++ b/omnibus/package-scripts/cinc/preinst
@@ -7,12 +7,12 @@
 #   this programming language.  do not touch.
 # - if you are under 40, get peer review from your elders.
 
-INSTALLER_DIR=/opt/chef
+INSTALLER_DIR=/opt/cinc
 if [ -e $INSTALLER_DIR ]; then
   echo "removing $INSTALLER_DIR..."
 
-  # have to do this dance of moving /opt/chef to a tmp dir since files may be in use
-  tmp_dir="/opt/.chef.$$"
+  # have to do this dance of moving /opt/cinc to a tmp dir since files may be in use
+  tmp_dir="/opt/.cinc.$$"
   # if we can't create the tmp_dir then fail hard to prevent any possible security hole
   (umask 077 && mkdir $tmp_dir) || exit 1
   # now we can clean up the tmp_dir we created safely
-- 
2.17.1

